src/main/java/org/folio/innreach/domain/service/impl/CirculationServiceImpl.java

```diff
@@ -262,6 +267,26 @@ public InnReachResponseDTO returnUncirculated(String trackingId, String centralC
     }
   }
 
+  @Override
+  public InnReachResponseDTO borrowerRenew(String trackingId, String centralCode, BorrowerRenewDTO borrowerRenew) {
+    var transaction = getTransaction(trackingId, centralCode);
+    var transactionState = transaction.getState();
+    var calculatedDueDate = transaction.getHold().getNeedBefore();
+    var requestedDueDate = borrowerRenew.getDueDateTime();
```

Comment:
> If **calculated due date** is < **requested due date**, or the renewal request fails and the **existing due date** is < the requested due date from the borrower

From this requirement it looks like there should be three due dates:
1) calculated due date
2) requested due date
3) existing due date

Can you please clarify with PO what each of them stands for? I assume that **requested due date** is the `dueDate` from borrowerRenew request but for the others I'm not sure. They may be either `needBefore` from transaction or `dueDate` from FOLIO Loan request

Reply:
There are 2 types of dates: from request and from loan